asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as grpc from '@grpc/grpc-js';
+import {
+    connect,
+    Contract,
+    Identity,
+    Signer,
+    signers,
+} from '@hyperledger/fabric-gateway';
+import * as crypto from 'crypto';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { TextDecoder } from 'util';
+
+const channelName = 'mychannel';
+const chaincodeName = 'private';
+const mspIdOrg1 = 'Org1MSP';
+const mspIdOrg2 = 'Org2MSP';
+
+// Path to org1 crypto materials.
+const cryptoPathOrg1 = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'test-network',
+    'organizations',
+    'peerOrganizations',
+    'org1.example.com'
+);
+
+// Path to org1 user private key directory.
+const keyDirectoryPathOrg1 = path.resolve(
+    cryptoPathOrg1,
+    'users',
+    'User1@org1.example.com',
+    'msp',
+    'keystore'
+);
+
+// Path to org1 user certificate.
+const certPathOrg1 = path.resolve(
+    cryptoPathOrg1,
+    'users',
+    'User1@org1.example.com',
+    'msp',
+    'signcerts',
+    'cert.pem'
+);
+
+// Path to org1 peer tls certificate.
+const tlsCertPathOrg1 = path.resolve(
+    cryptoPathOrg1,
+    'peers',
+    'peer0.org1.example.com',
+    'tls',
+    'ca.crt'
+);
+
+// Path to org2 crypto materials.
+const cryptoPathOrg2 = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'test-network',
+    'organizations',
+    'peerOrganizations',
+    'org2.example.com'
+);
+
+// Path to org2 user private key directory.
+const keyDirectoryPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'keystore'
+);
+
+// Path to org2 user certificate.
+const certPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'signcerts',
+    'cert.pem'
+);
+
+// Path to org2 peer tls certificate.
+const tlsCertPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'peers',
+    'peer0.org2.example.com',
+    'tls',
+    'ca.crt'
+);
+
+// Gateway peer endpoint.
+const peerEndpointOrg1 = 'localhost:7051';
+const peerEndpointOrg2 = 'localhost:9051';
+
+// Gateway peer container name.
+const peerNameOrg1 = 'peer0.org1.example.com';
+const peerNameOrg2 = 'peer0.org2.example.com';
```

Comment:
It would be great to move all this connection-specific stuff (including the connection helper functions, like `newGrpcConnection()`) to a separate file, as we did with the events sample. Only the basic sample is really demonstrating how to connect a Gateway. For these other samples, this just adds a lot of extra content that makes it much harder to focus on the bits this sample is really trying to demonstrate.